test/collect_coverage_api_test.dart

```diff
@@ -83,10 +83,38 @@ void main() {
     _expectHitCount(hits, 11, 1);
     _expectHitCount(hits, 28, 1);
   });
+
+  test('collect_coverage_api with function coverage', () async {
+    final json = await _collectCoverage(functionCoverage: true);
+    expect(json.keys, unorderedEquals(<String>['type', 'coverage']));
+    expect(json, containsPair('type', 'CodeCoverage'));
+
+    final coverage = json['coverage'] as List;
+    expect(coverage, isNotEmpty);
+
+    final sources = coverage.cast<Map>().fold(<String, List<Map>>{},
+        (Map<String, List<Map>> map, value) {
+      final sourceUri = value['source'] as String;
+      map.putIfAbsent(sourceUri, () => <Map>[]).add(value);
+      return map;
+    });
+
+    for (var sampleCoverageData in sources[_sampleAppFileUri]!) {
+      expect(sampleCoverageData['funcNames'], isNotNull);
+      expect(sampleCoverageData['funcHits'], isNotNull);
+    }
+
+    for (var sampleCoverageData in sources[_isolateLibFileUri]!) {
+      expect(sampleCoverageData['funcNames'], isNotEmpty);
```

Comment:
Why do we check for non-null above and non-empty here?

Reply:
Not sure. I was just copying `test('collect_coverage_api', ...`. Switched to isNotEmpty in both tests.